### PR TITLE
toolchain description update

### PR DESCRIPTION
**Pull request by mahdanoura**
addresses the second point in #1920

### CODE REVIEW

File: toolchain/README.md [hunk 1]
```diff
@@ -4,9 +4,15 @@
...
+
+The TD specification include diagrams that are also generated from the same SHACL and merged context sources. The STTL Engine is run once more to generated diagrams in textual form (DOT text files) and then turned into graphics using Graphviz tool. The output is the SVG figures of the TD specification, which is then manually converted into PNG using a desired tool of choice. These PNG files are used to feed the HTML specification. 
+
+The Parser Creation process is mainly performed by a developer using a 2-step process. The grammar is developed manually using to context-free grammar according to the semantics of STTL language. Then, Jison tool takes the grammar as input and outputs a JavaScript file capable of parsing the language described by that grammar. The generated script is then used to parse visualization template inputs in the STTL Engine.
```
Comment (egekorkan) [inline toolchain/README.md:18]:
```suggestion
The Parser Creation process is mainly performed by a developer using a 2-step process. The grammar is developed manually using a context-free grammar according to the semantics of STTL language. Then, the [Jison](https://gerhobbelt.github.io/jison/docs/) tool takes the grammar as input and outputs a JavaScript file capable of parsing the language described by that grammar. The generated script is then used to parse visualization template inputs in the STTL Engine.
```

**Review by egekorkan — CHANGES_REQUESTED** [review]

**Comment by egekorkan** [comment]
Another point from today's call:
- We should change the diagram's semantic web expert to "TD Task Force Participants" and write that in the readme that it is the main manual step.

File: toolchain/README.md [hunk 2]
```diff
@@ -4,9 +4,15 @@
 The W3C WoT specification generation process toolchain is depicted as a Business Process Modeling Notation diagram. There are 4 actors involved:
 1. ***Semantic Web Expert***
 2. ***WoT Toolchain***
```
Comment (mahdanoura) [inline toolchain/README.md:6]:
```suggestion
2. ***Render Script***
```

File: toolchain/README.md [hunk 3]
```diff
@@ -4,9 +4,15 @@
...
 
-The WoT Toolchain is a shell script (render.sh), which manages the interaction between different Node.js scripts. It receives the different JSON-LD context files for each schema, aggregates them into one and converts them into Turtle representation format. 
\ No newline at end of file
+The WoT Toolchain is a shell script (render.sh), which manages the interaction between different Node.js scripts developed to achieve different objectives. It receives the different JSON-LD context files for each vocabulary in the TD model and aggregates them into one. The resulting JSON-LD context is then converted into RDF Turtle representation format. The artifacts provided by the Semantic Web Experts and the RDF graphs are provided as input to the STTL Engine.
```
Comment (mahdanoura) [inline toolchain/README.md:12]:
```suggestion
The Render Script[https://github.com/w3c/wot-thing-description/blob/main/render.sh](url) is a shell script which manages the interaction between different Node.js scripts developed to achieve different objectives. First, it receives the different JSON-LD context files for each ontology in the TD model developed in the previous steps and aggregates them into one, resulting in [td-context-1.1.jsonld](https://github.com/w3c/wot-thing-description/blob/main/context/td-context-1.1.jsonld). The JSON-LD context file is then converted into RDF Turtle representation format ([td-context.ttl](https://github.com/w3c/wot-thing-description/blob/main/context/td-context.ttl)).
```

File: toolchain/README.md [hunk 4]
```diff
@@ -4,9 +4,15 @@
...
...rent RDF graphs: TD model ontologies, SHACL shapes (validation/td-validation.ttl), merged context (context/td-context.ttl), a transformation (a set of templates), and a specific starting node that determines the starting HTML node pointing to the WoT Things Descriptions as a set of command-line arguments. The TD model ontologies and SHACL shapes are loaded and stored in an RDF store for semantic processing and querying. Then, the STTL templates are parsed according to the parser developed by the Parser Creation actor. This is followed by matching the input HTML node with one of the transformation names specified in the template. A template in a transformation may apply other templates or include different utility functions or SPARQL functions. Therefore, the Apply Template sub-process identifies these different cases and applies the rules which includes the following activities: evaluate patterns, variables and expressions in a template, construct the query to execute using a JSON format, and executing the query on the RDF graph. The output is generated HTML textual output format corresponding to the index.html file of the TD specification and the different ontology specifications.
```
Comment (mahdanoura) [inline toolchain/README.md:14]:
```suggestion
The STTL engine takes different artifacts produced in the previous steps as input, including TD model ontologies, [SHACL shapes](https://github.com/w3c/wot-thing-description/blob/main/validation/td-validation.ttl), [merged context](https://github.com/w3c/wot-thing-description/blob/main/context/td-context-1.1.jsonld), a set of STTL transformation templates, and a specific starting node. The starting node is the HTML node pointing to the WoT Things Description file. The TD model ontologies and SHACL shapes are loaded and stored in an RDF store for semantic processing and querying. Then, the STTL templates are parsed according to the parser developed by the Parser Creation actor. This is followed by matching the input HTML node with one of the transformation names specified in the template. A template in a transformation may apply other templates or include different utility functions or SPARQL functions. Therefore, the Apply Template sub-process identifies these different cases and applies the rules which includes the following activities: evaluate patterns, variables and expressions in a template, construct the query to execute using a JSON format, and executing the query on the RDF graph. The output is generated HTML textual output format corresponding to the index.html file of the TD specification and the different ontology specifications.
```

[the remark this answers is not in the document]
Comment (mahdanoura) [reply toolchain/README.md:14]: Will create a PR for this for a second iteration

File: toolchain/README.md [hunk 5]
```diff
@@ -4,9 +4,15 @@
...
 
-The Semantic Web Expert has the main responsibility to iteratively define the schema of different vocabularies such as TD, TM, WoTsec, HCTL and any extensions required, as well as the JSON Schema definition and SHACL shapes for validation. These different tasks are performed in an adhoc manner until agreement is achieved among the experts. In addition, they define a set of templates for the specification according to the SPARQL Template Transformation Language (STTL).
+The Semantic Web Expert has the main responsibility to iteratively define the schema of different TD model specifications such as TD core, TM, WoTsec, HCTL or any extensions required. Also, the JSON Schema definition is defined for syntacticAL validation of JSON-LD data and Shapes Constraint Language (SHACL) shapes for Validating RDF graphs. These different tasks are performed in an adhoc manner until agreement is achieved among the experts. In addition, the experts define a set of transformations for the specification generation according to the SPARQL Template Transformation Language (STTL). The templates expressed in STTL describe different rules for transforming a source RDF graph into a result text.
```
Comment (egekorkan) [inline toolchain/README.md:10]:
```suggestion
The Semantic Web Expert has the main responsibility to iteratively define the schema of different TD model specifications such as TD core, TM, WoTsec, HCTL or any extensions required. Also, the JSON Schema definition is defined for syntactical validation of JSON-LD data and Shapes Constraint Language (SHACL) shapes for validating RDF graphs. These different tasks are performed in an adhoc manner until agreement is achieved among the experts. In addition, the experts define a set of transformations for the specification generation according to the SPARQL Template Transformation Language (STTL). The templates expressed in STTL describe different rules for transforming a source RDF graph into a result text.
```

**Review by egekorkan — CHANGES_REQUESTED** [review]
minor formatting issues

**Review by egekorkan — APPROVED** [review]

**Comment by mjkoster** [comment]
Meeting 1/17/2024
- Merge now and refine:
- Add a high level diagram
- Add detail diagrams as needed
- Add explanatory text as needed

**Pull request merged**